malloc/malloc.c

```diff
@@ -13,7 +14,13 @@
 */
 char *string_dup(char *src)
 {
+    int length = string_length(src);
+    char *copy = malloc(length+1); // why length + 1? 
```

Comment:
It might be `length + 1` to accommodate the null terminating character `'\0'` all strings should have.